src/Symfony/Component/Lock/Store/FlockStore.php

```diff
@@ -0,0 +1,142 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Lock\Store;
+
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\Lock\Exception\InvalidArgumentException;
+use Symfony\Component\Lock\Exception\LockConflictedException;
+use Symfony\Component\Lock\Exception\LockStorageException;
+use Symfony\Component\Lock\Key;
+use Symfony\Component\Lock\StoreInterface;
+
+/**
+ * FlockStore is a StoreInterface implementation using the FileSystem flock.
+ *
+ * Original implementation in \Symfony\Component\Filesystem\LockHandler.
+ *
+ * @author Jérémy Derussé <jeremy@derusse.com>
+ * @author Grégoire Pineau <lyrixx@lyrixx.info>
+ * @author Romain Neutron <imprec@gmail.com>
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+class FlockStore implements StoreInterface, BlockingStoreInterface
+{
+    private $lockPath;
+
+    /**
+     * @param string|null $lockPath The directory to store the lock. Default values will use temporary directory
+     *
+     * @throws LockStorageException If the lock directory could not be created or is not writable
+     */
+    public function __construct($lockPath = null)
+    {
+        $lockPath = $lockPath ?: sys_get_temp_dir();
+        if (!is_dir($lockPath)) {
+            $fs = new Filesystem();
+            try {
+                $fs->mkdir($lockPath);
```

Comment:
I don't think this belongs in the constructor to be honest, would prefer this in a cache warmer or something